### PR TITLE
Feature/add checkbox boolean question

### PR DESCRIPTION
**Pull request by josemanuel167**
Add render mode to boolean questions.

RADIO_BUTTONS and CHECKBOXES

### CODE REVIEW

**Comment by alextanlan** [comment]
It would be nice to have some screenshots as to see how it looks like.

**Comment by alextanlan** [comment]
It makes sense to rename `boolean` directory to `activity-boolean-answer` in order to be consistent with others, like `activity-file-answer` , `activity-matrix-answer`.

**Comment by alextanlan** [comment]
Generally, having two child components (`ddp-activity-boolean-radio-buttons-answer` / `ddp-activity-boolean-checkbox-answer`) seems like an overkill, imo. They duplicate each other much. They are just two views of the same component, but the logic is practically the same.

My proposal is to keep origin `ActivityBooleanAnswer` component and tweak its template according to a `renderMode`. 
Using either `<ng-template>` or just `<ng-container *ngIf='some condition'>` or even  [conditionally Changing the Template File](https://medium.com/ngconf/conditional-templates-for-angular-components-25e378fb6eb5).

**Comment by alextanlan** [comment]
Generally, having two child components (`ddp-activity-boolean-radio-buttons-answer` / `ddp-activity-boolean-checkbox-answer`) seems like an overkill, imo. They duplicate each other much. They are just two views of the same component, but the logic is practically the same.

My proposal is to keep origin `ActivityBooleanAnswer` component and tweak its template according to a `renderMode`. 
Using either `<ng-template>` or just `<ng-container *ngIF='some condition'>` or even  [conditionally Changing the Template File](https://medium.com/ngconf/conditional-templates-for-angular-components-25e378fb6eb5).

**Comment by alextanlan** [comment]
Tests failed in the branch (see [CIrckeCI workflow)](https://app.circleci.com/pipelines/github/broadinstitute/ddp-angular/14733/workflows/599a0169-7994-4bc5-83ea-669466b1785b/jobs/22354).

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { ActivityBooleanQuestionBlock } from '../../../../../models/activity/activityBooleanQuestionBlock';
+
+@Component({
+    selector: 'ddp-activity-boolean-checkbox-answer',
+    template: `
+        {{ block.answer }}
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts:7]: Do we need it here ? 
it is just **true** / **false**.

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        <mat-checkbox
+            [checked]="block.answer"
+            [disabled]="readonly"
+            [disableRipple]="true"
+            (change)="onValueChange()"
+        >
+            <p
+                [innerHTML]="
+                    block.answer ? block.trueContent : block.falseContent
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts:16]: I suppose, the component CHECKBOX view should not use **trueContent**/**falseContent** properties at all.

File: ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityBooleanQuestionBlock.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import { QuestionType } from './questionType';
 export class ActivityBooleanQuestionBlock extends ActivityQuestionBlock<boolean> {
     public trueContent: string;
     public falseContent: string;
+    public renderMode: string;
```
Comment (alextanlan) [inline ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityBooleanQuestionBlock.ts:7]:
```
public renderMode: BooleanRenderMode;
```

**Comment by josemanuel167** [comment]
> It makes sense to rename `boolean` directory to `activity-boolean-answer` in order to be consistent with others, like `activity-file-answer` , `activity-matrix-answer`.



> It makes sense to rename `boolean` directory to `activity-boolean-answer` in order to be consistent with others, like `activity-file-answer` , `activity-matrix-answer`.

@alextanlan I was following picklist structure which is the one I'm familiar with, picklist folder is named "Picklist" only

File: ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityBooleanQuestionBlock.ts (hunk 3 above)
Comment (josemanuel167) [reply ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityBooleanQuestionBlock.ts:7]: thanks

Comment (josemanuel167) [reply ddp-workspace/projects/ddp-sdk/src/lib/models/activity/activityBooleanQuestionBlock.ts:7]: thanks

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts (hunk 2 above)
Comment (josemanuel167) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts:16]: Sounds good

File: ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts (hunk 1 above)
Comment (josemanuel167) [reply ddp-workspace/projects/ddp-sdk/src/lib/components/activityForm/answers/boolean/activity-boolean-checkbox-answer/activity-boolean-checkbox-answer.component.ts:7]: Nope, my bad, I forgot to delete

**Comment by alextanlan** [comment]
The implementation was made/ continued in [PR#1202](https://github.com/broadinstitute/ddp-angular/pull/1202)

**Pull request closed without merging**